content/docs/pipelines/sdk/python-based-visualizations.md

```diff
@@ -0,0 +1,105 @@
+customers to provided Python visualization code to be used to generate
+visualizations. These visualizations allow for rapid development,
+experimentation, and customization when visualizing results.
+
+<img src="/docs/images/python-based-visualizations1.png" 
+  alt="Confusion matrix visualization from a pipeline component"
+  class="mt-3 mb-3 border border-info rounded">
+
+## Using predefined visualizations
+
+<img src="/docs/images/python-based-visualizations2.png" 
+  alt="Confusion matrix visualization from a pipeline component"
+  class="mt-3 mb-3 border border-info rounded">
+
+1. Open the details of a run.
+2. Select a component.
+    * The component that is selected does not matter. But, if you want to
+    visualize the output of a specific component, it is easier to do that within
+    that component.
+3. Select the **Artifacts** tab.
+4. At the top of the tab you should see a card named **Visualization Creator**.
+5. Within the card, provide a visualization type, a source, and any necessary
+arguments.
+    * Any required or optional arguments will be shown as a placeholder.
+6. Click **Generate Visualization**.
+7. View generated visualization by scrolling down.
+
+## Using custom visualizations
+
+<img src="/docs/images/python-based-visualizations3.png" 
+  alt="Confusion matrix visualization from a pipeline component"
+  class="mt-3 mb-3 border border-info rounded">
+
+1. Enable custom visualizations within Kubeflow Pipelines.
+    * If you have not yet deployed Kubeflow Pipelines to your cluster,
+    you can edit the [frontend deployment YAML](https://github.com/kubeflow/
+    pipelines/blob/master/manifests/kustomize/base/pipeline/
+    ml-pipeline-ui-deployment.yaml)
+    file to include the following YAML that specifies that custom visualizations
+    are allowed via environment variables.
+
+    ```YAML
+    - env:
+      - name: ALLOW_CUSTOM_VISUALIZATIONS
+        value: true
+    ```
+    * If you already have Kubeflow Pipelines deployed within a cluster, you can
+    edit the frontend deployment YAML to specify that custom visualizations are
+    allowed in the same way described above. Details about updating
+    deployments can be found in the Kubernetes documentation about
+    [updating a deployment](https://kubernetes.io/docs/concepts/workloads/
+    controllers/deployment/#updating-a-deployment).
+2. Open the details of a run.
+3. Select a component.
+    * The component that is selected does not matter. But, if you want to
+    visualize the output of a specific component, it is easier to do that within
+    that component.
+4. Select the **Artifacts** tab.
+5. At the top of the tab you should see a card named **Visualization Creator**.
+6. Within the card, select the **CUSTOM** visualization type then provide a
+source, and any necessary arguments (the source and argument variables are
+optional for custom visualizations).
+7. Provide the custom visualization code.
```

Comment:
We should include some guidelines about what the Python code looks like. Recommendation: Copy any relevant bits from [the contributor README](https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/visualization/README.md), including:

* The requirement for Python 3.
* How to pass variables from Pipelines to the UI.
* Anything else that the Python developer needs to know - please consult the engineering team for these details.
* An example of the code for a simple visualization.

Reply:
Still need to address:
* Anything else that the Python developer needs to know - please consult the engineering team for these details.
* An example of the code for a simple visualization.